### PR TITLE
Added identity and occurrences to evidence. Updated test cases.

### LINKED ISSUE #129: Add evidence used to determine inclusion of a component.
Most SBOM generators base the inclusion of a component in an SBOM on a packagemanager file or the existence of some other file.
I would like to be able to trace back what source was used as evidence for the inclusion of a component. 
The `component.evidence` property seemed like a good fit, but that only supports "licenses" and "copyright".

Could we add a "files" list to that?
Then an SBOM generator can list the file(s) it used to decide to include the component there,

### PR DESCRIPTION
**Pull request by stevespringett**
Closes #129

### CODE REVIEW

File: tools/src/test/resources/1.5/valid-evidence-1.5.json [hunk 1]
```diff
@@ -19,6 +19,40 @@
       ],
       "purl": "pkg:maven/com.google.code.findbugs/findbugs-project@3.0.0",
       "evidence": {
+        "identity": {
+          "field": "purl",
```
Comment (stevespringett) [inline tools/src/test/resources/1.5/valid-evidence-1.5.json:23]: I'm not sold on the idea that this "field" needs to even be there. The original idea was to be able to separate out individual evidence of group/name/version into individual pieces of evidence. But for some things, it actually doesn't make sense (like purl).

Comment (stevespringett) [reply tools/src/test/resources/1.5/valid-evidence-1.5.json:23]: Same would apply fo "field" in the XML and protobuf.

**Comment by jkowalleck** [comment]
according to https://github.com/CycloneDX/specification/blob/56dfd3dbd957dab6ce4d48310b9ec70c8a21ef8d/schema/bom-1.5.schema.json#L1262-L1268
and https://github.com/CycloneDX/specification/blob/56dfd3dbd957dab6ce4d48310b9ec70c8a21ef8d/schema/bom-1.5.schema.json#L1299-L1305

We have an overall confidence, and we have multiple specific confidence values.
One could argue that an overall confidence might be derived of (e.g. meridian over all) all specific confidences. This would be true if all specifics were weighted equally. 
The mere existence of an overall confidence value indicates that this is not the case.

Therefore, i would argue, that the weighing of each specific confidence value should be published.
And if this was published, then an overall value would no longer make sense, since it could be calculated by each tool that digested this document.

:x: Therefore, Ii do not agree to the existence of such property.

**Comment by stevespringett** [comment]
@jkowalleck Thanks for your input. Based on today's IWG meeting, it's my understanding that `weight` is no longer being considered, as its up to the consumer to place weight on what they believe are the most important techniques for analysis. Is  that correct? Is weight being withdrawn? If so, then perhaps some documentation improvements to describe consumer weighting?

**Comment by stevespringett** [comment]
Per https://github.com/CycloneDX/specification/issues/129#issuecomment-1488483425 callstack is being added in. Will update when complete.

**Comment by stevespringett** [comment]
Ok, callstack support has been added back into this PR and test cases

**Pull request merged**